packages/@sanity/desk-tool/src/panes/document/documentPanel/header/DocumentPanelHeader.tsx

```diff
@@ -18,118 +18,125 @@ import {DocumentHeaderTitle} from './DocumentHeaderTitle'
 
 export interface DocumentPanelHeaderProps {
   rootElement: HTMLDivElement | null
+  isReferencedDocument?: boolean
 }
 
 const isActionButton = (item: PaneMenuItem) => Boolean(item.showAsAction)
 const isMenuButton = negate(isActionButton)
 
 export const DocumentPanelHeader = memo(
-  forwardRef(({rootElement}: DocumentPanelHeaderProps, ref: React.ForwardedRef<HTMLDivElement>) => {
-    const {
-      documentSchema,
-      handleMenuAction,
-      handlePaneClose,
-      handlePaneSplit,
-      historyController,
-      markers,
-      menuItems,
-      menuItemGroups,
-      ready,
-      views,
-    } = useDocumentPane()
-    const {revTime: rev} = historyController
-    const {features} = useDeskTool()
-    const {index, BackLink, hasGroupSiblings} = usePaneRouter()
-    const contextMenuItems = useMemo(() => menuItems.filter(isMenuButton), [menuItems])
-    const [isValidationOpen, setValidationOpen] = React.useState<boolean>(false)
-    const showTabs = views.length > 1
-    const showVersionMenu = features.reviewChanges
+  forwardRef(
+    (
+      {rootElement, isReferencedDocument}: DocumentPanelHeaderProps,
+      ref: React.ForwardedRef<HTMLDivElement>
+    ) => {
```

Comment:
This is very nitpicky and only addressing Victorias note above - but theoretically I think you might force prettier to keep formatting if you do something like this:

```suggestion
  forwardRef((headerProps: DocumentPanelHeaderProps, ref: React.ForwardedRef<HTMLDivElement>) => {
  const {rootElement, isReferencedDocument} = headerProps
```